src/libDirectoryService/ViewChangePostProcessing.cpp

```diff
@@ -322,11 +322,18 @@ void DirectoryService::ProcessViewChangeConsensusWhenDone() {
       t_microBlocks.emplace(microBlock.GetHeader().GetShardId(), microBlock);
     }
 
+    DequeOfShard t_shards;
+    if (m_forceMulticast && GUARD_MODE) {
+      ReloadGuardedShards(t_shards);
+    }
+
     DataSender::GetInstance().SendDataToOthers(
-        *m_pendingVCBlock, tmpDSCommittee, m_shards, t_microBlocks,
+        *m_pendingVCBlock, tmpDSCommittee,
+        (m_forceMulticast && GUARD_MODE) ? t_shards : m_shards, t_microBlocks,
```

Comment:
Maybe can be changed like following, to avoid multiple comparison

   DequeOfShard t_shards;
  if (m_forceMulticast && GUARD_MODE) {
    ReloadGuardedShards(t_shards);
  }
  else {
    t_shards = m_shards;
  }

   DataSender::GetInstance().SendDataToOthers(
      *m_finalBlock, *m_mediator.m_DSCommittee, m_shards, t_microBlocks,